src/test/unit_tests/jest_tests/test_nc_upgrade_manager.test.js

```diff
@@ -673,6 +726,101 @@ describe('nc upgrade manager - upgrade config directory', () => {
+            const res = await nc_upgrade_manager.upgrade_config_dir(expected_version, hosts_list);
+            const system_json_data = await config_fs.get_system_config_file();
+            const expected_data = {
+                ...system_data,
+                config_directory: {
+                    phase: CONFIG_DIR_PHASES.CONFIG_DIR_UNLOCKED,
+                    config_dir_version: config_fs.config_dir_version,
+                    upgrade_package_version: expected_version,
+                    upgrade_history: {
+                        successful_upgrades: [res]
+                    }
+                }
+            };
+            assert_upgrade_finish_or_fail_data(system_json_data, expected_data);
+        });
+
+        it('upgrade_config_dir - system.json and package.json have the same version', async () => {
+            const system_data = _.cloneDeep(old_expected_system_json_has_config_directory);
+            const expected_version = pkg.version;
+            const hosts_list = [hostname];
+            await config_fs.create_system_config_file(JSON.stringify(system_data));
+            const expected_msg = { "message": "config_dir_version on system.json and config_fs.config_dir_version match, nothing to upgrade" };
+            const res = await nc_upgrade_manager.upgrade_config_dir(expected_version, hosts_list);
+            expect(res).toStrictEqual(expected_msg);
+        });
+
+        it('upgrade_config_dir - config_dir_version in system.json is newer than the hosts current config_dir_version', async () => {
+            const system_data = _.cloneDeep(newer_expected_system_json_has_config_directory);
+            const expected_version = pkg.version;
+            const hosts_list = [hostname];
+            await config_fs.create_system_config_file(JSON.stringify(system_data));
+            const expected_err_msg = 'attempt to run old container version with newer server version';
+            await expect(nc_upgrade_manager.upgrade_config_dir(expected_version, hosts_list))
+                .rejects.toThrow(expected_err_msg);
+        });
+
+        it('upgrade_config_dir - host RPM is not upgraded to the expectced version', async () => {
+            const system_data = _.cloneDeep(old_expected_system_json);
+            const expected_version = pkg.version;
+            const hosts_list = [hostname];
+            await config_fs.create_system_config_file(JSON.stringify(system_data));
+            const expected_err_msg = `config dir upgrade can not be started until all expected hosts have the expected version=${pkg.version}, host=${hostname} host's current_version=${system_data[hostname].current_version}`;
+            await expect(nc_upgrade_manager.upgrade_config_dir(expected_version, hosts_list))
+                .rejects.toThrow(expected_err_msg);
+        });
+
+        it('upgrade_config_dir - old successful upgrades is not empty', async () => {
```

Comment:
Could you comment on which step this should be cleaned (as it is information about the previous upgrade).

Reply:
I didn't understand this comment, what should be cleaned? successful upgrades are never cleaned-up only the last_failure